### ISSUE TITLE
Undo does not work in Notebook Cells (confusion around cell level undo/redo stack)

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#133026]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: Insiders
- OS Version: Mac

Steps to Reproduce:

1. Open an existing ipynb notebook
2. Copy the last cell down (duplicate it)
3. Hit the `z` key (it should get deleted)
4. This works
5. Next duplicate the cell again & click else where in the notebook (empty area below the last cell)
6. Now set focus back to the last cell and hit `z`
7. Nothing happens
8. From now on, `z` will never work, `undo` command from the command palette too doesn't work

I've tried enabling/disabling the setting `Cell level undo/redo stack`, no difference.
I think there's a confusion (discussed this in nb sync meeting) around what the `Cell Level undo/redo stack` does as well (or perhaps the confusion is due to the fact that this doesn't seem to work).

TLDR: I wasn't able to get undo/redo working reliably enough (in fact majority of the time it didn't work as expected).

**Comment by nickeubank** [comment]
I just want to flag this is something that I think is pretty important to address -- its **really** easy to lose a lot of work if an errant delete command deletes the wrong cell and there's no undo.

**Comment by roblourens** [comment]
This might have been https://github.com/microsoft/vscode/issues/131937

**Comment by rebornix** [comment]
@DonJayamanne can you help confirm if what you are seeing is the same as #131937?

**Comment by wwdok** [comment]
I have also found that i can only undo those operations in one cell, not the whole notebook, moreover, if i delete one cell, i can not undo it.

**Comment by shanecarlson** [comment]
Just lost a few hours work of work due to this bug. A fix would be nice.

**Comment by DonJayamanne** [comment]
I'm closing this issue as I can no longer replicate this issue.
